snuba/settings/validation.py

```diff
@@ -117,3 +117,48 @@ def validate_slicing_settings(locals: Mapping[str, Any]) -> None:
         assert (
             topic_tuple in locals["SLICED_KAFKA_BROKER_CONFIG"]
         ), f"missing broker config definition for sliced Kafka topic {topic_tuple[0]} on slice {topic_tuple[1]}"
+
+    _STORAGE_SET_CLUSTER_MAP: Dict[str, Mapping[str, Any]] = {}
+
+    _SLICED_STORAGE_SET_CLUSTER_MAP: Dict[Tuple[str, int], Mapping[str, Any]] = {}
+    for cluster in locals["CLUSTERS"]:
+        for storage_set in cluster["storage_sets"]:
+            _STORAGE_SET_CLUSTER_MAP[storage_set] = cluster
```

Comment:
this is clear so it's fine but i think it could be done as a list comprehension and then you wouldn't need a mutable variable type:

```python
_STORAGE_SET_CLUSTER_MAP = { 
  storage_set: cluster 
  for storage_set in cluster["storage_sets"] 
  for cluster in locals["CLUSTERS"] 
}
```